ref-exchange/src/owner.rs

```diff
@@ -15,18 +16,58 @@ impl Contract {
         self.owner_id.clone()
     }
 
+    /// Extend guardians. Only can be called by owner.
+    #[payable]
+    pub fn extend_guardians(&mut self, guardians: Vec<ValidAccountId>) {
+        self.assert_owner();
+        for guardian in guardians {
+            self.guardians.insert(guardian.as_ref());
+        }
+    }
+
+    /// Remove guardians. Only can be called by owner.
+    pub fn remove_guardians(&mut self, guardians: Vec<ValidAccountId>) {
+        self.assert_owner();
+        for guardian in guardians {
+            self.guardians.remove(guardian.as_ref());
+        }
+    }
+
+    /// Change state of contract, Only can be called by owner or guardians.
+    #[payable]
+    pub fn change_state(&mut self, state: RunningState) {
+        assert_one_yocto();
+        assert!(self.is_owner_or_guardians(), "ERR_NOT_ALLOWED");
+        let cur_state: String = (&self.state).into();
+        let new_state: String = (&state).into();
+        if new_state != cur_state {
```

Comment:
You'll be able to compare them directly, once you derive `PartialEq` or `RunningState`